hazelcast/aggregator.py

```diff
@@ -89,245 +88,373 @@ def get_class_id(self):
+        input values.
     """
-    return _MinAggregator(attribute_path)
+    return _IntegerSumAggregator(attribute_path)
 
 
-def int_avg(attribute_path=None):
-    """Creates int average aggregator.
+def long_avg(attribute_path=None):
+    """Creates an aggregator that calculates the average of the input values.
 
-    Does NOT accept ``None`` input values.
-    Accepts only long input values (primitive and boxed).
+    Does NOT accept ``None`` input values or ``None`` extracted values.
+
+    Since the server-side implementation is in Java, values stored in the
+    Map must be of type ``long`` (primitive or boxed) in Java or of a type
+    that can be converted to that. That means, one should be able to use this
+    aggregator with ``int`` values sent from the Python client unless they
+    are out of range for ``long`` type in Java.
 
     Args:
-        attribute_path (str): extracts values from this path if given
+        attribute_path (str): Extracts values from this path, if given.
 
     Returns:
-        Aggregator[int]: an aggregator that calculates the average of the
-        input values
+        Aggregator[int]: An aggregator that calculates the average of the
+        input values.
     """
-    return _IntegerAverageAggregator(attribute_path)
+    return _LongAverageAggregator(attribute_path)
 
 
-def int_sum(attribute_path=None):
-    """Creates int sum aggregator.
+def long_sum(attribute_path=None):
+    """Creates an aggregator that calculates the sum of the input values.
 
-    Does NOT accept ``None`` input values.
-    Accepts only long input values (primitive and boxed).
+    Does NOT accept ``None`` input values or ``None`` extracted values.
+
+    Since the server-side implementation is in Java, values stored in the
+    Map must be of type ``long`` (primitive or boxed) in Java or of a type
+    that can be converted to that. That means, one should be able to use this
+    aggregator with ``int`` values sent from the Python client unless they
+    are out of range for ``long`` type in Java.
 
     Args:
-        attribute_path (str): extracts values from this path if given
+        attribute_path (str): Extracts values from this path, if given.
 
     Returns:
-        Aggregator[int]: an aggregator that calculates the sum of the
-        input values
+        Aggregator[int]: An aggregator that calculates the sum of the
+        input values.
     """
-    return _IntegerSumAggregator(attribute_path)
+    return _LongSumAggregator(attribute_path)
 
 
-def long_avg(attribute_path=None):
-    """Creates long average aggregator.
+def max_(attribute_path=None):
+    """Creates an aggregator that calculates the max of the input values.
 
-    Does NOT accept ``None`` input values.
-    Accepts only long input values (primitive and boxed).
+    Accepts ``None`` input values and ``None`` extracted values.
+
+    Accepts any input value type that is comparable.
```

Comment:
We mentioned java types etc. in other docstrings. To make this `comparable` word more explicit, should we mention Comparable interface of Java? Same for min, max_by, min_by.

Reply:
Improved the documentation for such aggregators. Please see the latest commit